### PR TITLE
ping_exporter: init at 1.1.0

### PR DESCRIPTION
**Pull request by Noodlesalat**
## Description of changes

This adds @czerwonk's [ping_exporter](https://github.com/czerwonk/ping_exporter) to Nixpkgs. ping_exporter is a "Prometheus exporter for ICMP echo requests".

A Grafana dashboard is also available via [https://grafana.com/grafana/dashboards/18194](https://grafana.com/grafana/dashboards/18194).

The commit messages are not fitting [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), because I did not know how to split them without giving them all the same message.

As this is my first contribution to nixpkgs, feedback is highly appreciated. :slightly_smiling_face:

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---
Thanks @mweinelt and @pearzt for their great support!

### CODE REVIEW

**Comment by marcusramberg** [comment]
The commit message for adding the module should probably be 
```
nixos/(module): init module 

(Motivation for change. Link to release notes. Additional information.)
``` 
and only the package should include the version. I think the tests should go in the same commit as the module as well, and since 23.11 is out now, release notes should probably go in the next release?

**Comment by mweinelt** [comment]
> I think the tests should go in the same commit as the module as well

No, they shouldn't.

**Comment by marcusramberg** [comment]
> > I think the tests should go in the same commit as the module as well
> 
> No, they shouldn't. Check existing commits on `nixos/tests/`.

Thanks, my bad on that one :)

**Review by Janik-Haag — COMMENTED** [review]
Looks pretty solid, thank you for your PR. This review is mostly nitpicking.

**Comment by Janik-Haag** [comment]
also please rename your commits to look like:
`maintainers: add Nudelsalat `
`prometheus-ping-exporter: init at 1.1.0` 
`nixos/prometheus-ping-exporter: init`
`nixos/tests/prometheus-ping-exporter: init`

and squash the release-notes commit into the `nixos/prometheus-ping-exporter: init`

File: nixos/modules/services/monitoring/prometheus/exporters/ping.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  # for convenience we define cfg here
```
[the remark this answers is not in the document]
Comment (Noodlesalat) [reply nixos/modules/services/monitoring/prometheus/exporters/ping.nix]: Comment removed :)

File: nixos/modules/services/monitoring/prometheus/exporters/ping.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  settingsFormat = pkgs.formats.yaml {};
+  configFile = settingsFormat.generate "config.yml" cfg.settings;
+in
+{
+  port = 9427; # The ping exporter listens on this port by default
+
+  # `extraOpts` is an attribute set which contains additional options
+  # (and optional overrides for default options).
+  # Note that this attribute is optional.
+  extraOpts = {
```
[the remark this answers is not in the document]
Comment (Noodlesalat) [reply nixos/modules/services/monitoring/prometheus/exporters/ping.nix]: Comment removed :)

File: nixos/modules/services/monitoring/prometheus/exporters/ping.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  # `serviceOpts` is an attribute set which contains configuration
+  # for the exporter's systemd service. One of
+  # `serviceOpts.script` and `serviceOpts.serviceConfig.ExecStart`
+  # has to be specified here. This will be merged with the default
+  # service configuration.
+  serviceOpts = {
+    serviceConfig = {
+      CapabilityBoundingSet = [ "CAP_NET_RAW" ];
+      AmbientCapabilities = [ "CAP_NET_RAW" ];
```
[the remark this answers is not in the document]
Comment (Noodlesalat) [reply nixos/modules/services/monitoring/prometheus/exporters/ping.nix]: Changes applied :)

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 4]
```diff
@@ -137,6 +137,8 @@
 
 - [c2FmZQ](https://github.com/c2FmZQ/c2FmZQ/), an application that can securely encrypt, store, and share files, including but not limited to pictures and videos. Available as [services.c2fmzq-server](#opt-services.c2fmzq-server.enable).
 
+- [ping_exporter](https://github.com/czerwonk/ping_exporter), a Prometheus exporter for ICMP echo requests. Available as [services.prometheus.exporters.ping](#opt-services.prometheus.exporters.ping.enable).
+
```
[the remark this answers is not in the document]
Comment (Noodlesalat) [reply nixos/doc/manual/release-notes/rl-2311.section.md]: Migrated to rl-2405

**Comment by Noodlesalat** [comment]
Thank you all for you input and sorry for the long delay! I think I incorporated all of your suggestions. Hopefully this is ready for merge right now. 🙂

Last but not least: Have a wonderful christmas time! :christmas_tree: :santa:

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**